pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -0,0 +1,252 @@
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	var err error
+	var alertData AMReceiverData
+	err = json.NewDecoder(r.Body).Decode(&alertData)
+	if err != nil {
+		log.Errorf("Failed to process request body: %s\n", err)
+		http.Error(w, "Bad request body", http.StatusBadRequest)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	// process request
+	response := h.processAMReceiver(alertData, r.Context())
+
+	// write response
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(response.Code)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		log.Errorf("Failed to write to response: %s\n", err)
+		http.Error(w, "Failed to write to response", http.StatusInternalServerError)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	metrics.ResetMetric(metrics.MetricRequestFailure)
+}
+
+func (h *WebhookRHOBSReceiverHandler) processAMReceiver(d AMReceiverData, ctx context.Context) *AMReceiverResponse {
+	log.WithField("AMReceiverData", fmt.Sprintf("%+v", d)).Info("Process alert data")
+
+	listOptions := []client.ListOption{
+		client.InNamespace(OCMAgentNamespaceName),
+	}
+	// Let's get all ManagedFleetNotifications
+	mfnl := &oav1alpha1.ManagedFleetNotificationList{}
+	err := h.c.List(ctx, mfnl, listOptions...)
+	if err != nil {
+		log.WithError(err).Error("unable to list managedFleetNotifications")
+		return &AMReceiverResponse{Error: err, Status: "unable to list managedFleetNotifications", Code: http.StatusInternalServerError}
+	}
+
+	// Handle each firing alert
+	for _, alert := range d.Alerts.Firing() {
+		// Filter actionable alert based on Label
+		if !actionableAlert(alert) {
+			continue
+		}
+		err = h.processAlert(alert, mfnl, true)
+		if err != nil {
+			log.WithError(err).Error("a firing alert could not be successfully processed")
+		}
+	}
+	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
+}
+
+// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// and returns an error if that process completed successfully or false otherwise
+func (h *WebhookRHOBSReceiverHandler) processAlert(alert template.Alert, mfnl *oav1alpha1.ManagedFleetNotificationList, firing bool) error {
+	// Should this alert be handled?
+	if !isValidAlert(alert) {
+		log.WithField(LogFieldAlert, fmt.Sprintf("%+v", alert)).Info("alert does not meet valid criteria")
+		return fmt.Errorf("alert does not meet valid criteria")
+	}
+
+	// Can the alert be mapped to an existing notification definition?
+	fn, mfn, err := getFleetNotification(alert.Labels[AMLabelTemplateName], mfnl)
+	if err != nil {
+		log.WithError(err).WithField(LogFieldAlert, fmt.Sprintf("%+v", alert)).Warning("an alert fired with no associated notification template definition")
+		return err
+	}
+
+	mcID := alert.Labels[AMLabelAlertMCID]
+	hcID := alert.Labels[AMLabelAlertHCID]
```

Comment:
It may be worth doing a check to ensure that both of these labels actually exist, and skip if they don't.

I'm also thinking of the cases where the hosted cluster ID label doesn't exist, just in case the fleet mode OA ends up getting alerts that are firing on the management cluster (which should be handled by the non-fleet mode OA running on the cluster)

Reply:
(This is now resolved)